provider/filestore.go

```diff
@@ -663,3 +663,24 @@ func (fs *HDFSFileStore) FilestoreType() pc.FileStoreType {
 func (fs *HDFSFileStore) AddEnvVars(envVars map[string]string) map[string]string {
 	panic("HDFS Filestore is not supported for K8s at the moment.")
 }
+
+/**
+ * Checks if the source url i.e "abfss://test@testingstoragegen.dfs.core.windows.net/featureform/tests/ice_cream.parquet" exists in the file store.
+ */
+func doesSourceUrlExist(store FileStore, sourceUrl string) (bool, error) {
```

Comment:
Is there a reason why this is a standalone function instead of part of the FileStore?

Reply:
It's a good question. I chose not to do for because I like the core interface functions on the interface. I don't want to bloat the interface with a ExistsByUrl since this feels more like a utility function and also since getting the file parts by url works only for azure, gcs, and s3 right now. I could be convinced otherwise but that was my current reasoning.

Reply:
i'm really 50/50 on it right now the more I think about it. Because everything is stored as a url.

Reply:
If this code would be exactly the same for each of them (which seems to be the case) then it should be a utility as you have here. My only nitpick on it is that the function name should be capitalized assuming its also meant to be used outside of this package.